src/main/java/net/sf/jabref/logic/integrity/EditionChecker.java

```diff
@@ -16,7 +16,7 @@
 public class EditionChecker implements Checker {
 
     private static final Predicate<String> FIRST_LETTER_CAPITALIZED = Pattern.compile("^[A-Z]").asPredicate();
-    private static final Predicate<String> ONLY_NUMERALS_OR_LITERALS = Pattern.compile("^([0-9]+|[^0-9]+)$")
+    private static final Predicate<String> ONLY_NUMERALS_OR_LITERALS = Pattern.compile("^([0-9]+|[^0-9].+)$")
```

Comment:
Literals are numbers, letters and strings. [^0-9]+ excludes numbers. This is an unintentinal restriction. The fullstop indicates, that all literals are allowed, except for the first character, that should not be a number (to allow "Edition 2000" and avoid something like "2nd").